### PR TITLE
Fix hotloop once the watch closes

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

See https://github.com/kubernetes/kubernetes/pull/126556#discussion_r1724984941

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/volume/persistent_volumes_test.go [hunk 1]
```diff
@@ -1068,7 +1068,11 @@ func TestPersistentVolumeProvisionMultiPVCs(t *testing.T) {
 		}
 		for {
 			select {
-			case event := <-w.ResultChan():
+			case event, ok := <-w.ResultChan():
+				if !ok {
+					t.Error("watch channel should not close")
```
Comment (liggitt) [inline test/integration/volume/persistent_volumes_test.go:1073]: probably want to just log here, not error, unless we're wanting to make the test fail in new ways

a watch closing is a normal thing, not an error

Comment (liggitt) [reply test/integration/volume/persistent_volumes_test.go:1073]: alternatively, you could log this happened and restart the watch

Comment (carlory) [reply test/integration/volume/persistent_volumes_test.go:1073]: Is it like this? @liggitt 

```go
		for {
			select {
			case event, ok := <-w.ResultChan():
				if !ok {
					klog.Info("Event watch channel closed")
					w, err = testClient.EventsV1().Events(ns.Name).Watch(tCtx, metav1.ListOptions{})
					if err != nil {
						return
					}
					continue
				}
				reportToArtifacts(t.Name()+"-events.text", event.Object)
			case <-tCtx.Done():
				w.Stop()
				return
			}
		}
```

Comment (carlory) [reply test/integration/volume/persistent_volumes_test.go:1073]: I'm not sure if we need to restart the watch

Comment (liggitt) [reply test/integration/volume/persistent_volumes_test.go:1073]: yeah, that looks reasonable if we're just debugging

Comment (carlory) [reply test/integration/volume/persistent_volumes_test.go]: updated

File: test/integration/volume/persistent_volumes_test.go [hunk 2]
```diff
@@ -1068,7 +1068,15 @@ func TestPersistentVolumeProvisionMultiPVCs(t *testing.T) {
 		}
 		for {
 			select {
-			case event := <-w.ResultChan():
+			case event, ok := <-w.ResultChan():
+				if !ok {
+					klog.Info("Event watch channel closed")
+					w, err = testClient.EventsV1().Events(ns.Name).Watch(tCtx, metav1.ListOptions{})
+					if err != nil {
+						return
```
Comment (liggitt) [inline test/integration/volume/persistent_volumes_test.go:1076]: log this error happened and the watch restart failed

Comment (carlory) [reply test/integration/volume/persistent_volumes_test.go:1077]: updated

**Pull request merged**